sound/soc/sof/ipc4-mtrace.c

```diff
@@ -128,6 +129,13 @@ static int sof_ipc4_mtrace_dfs_open(struct inode *inode, struct file *file)
 	struct sof_mtrace_core_data *core_data = inode->i_private;
 	int ret;
 
+	mutex_lock(&core_data->buffer_lock);
+
+	if (core_data->log_buffer) {
+		ret = -EBUSY;
+		goto out;
```

Comment:
We should `goto out` whenever in error, right? Like line 141, 146

Reply:
@bardliao uhm yes, will fix,